File: API_R_01Agosto2024/Joao victor de sousa dos santos/API-R#01/q3_cidades.js
// q3_cidades

var input = require("readline-sync");

function main(){

    // entrada
    var quantidade_cidades = input.questionInt("\n> Quantidade de cidades: ")

    // processamento
    var renda_media = 0;
    var menor_renda = 0;
    var maior_renda = 0;
    var somatorio_de_vendas = 0;
    var contador = 1;
    var renda_cidade_atual = 1;

    while (contador <= quantidade_cidades){
        renda_cidade_atual = input.questionFloat(`> renda cidade #${contador} R$: `)

        if (renda_cidade_atual == 0){
            break;
        }

        // para o primeiro loop
        if (contador == 1){
            menor_renda = renda_cidade_atual;
            maior_renda = renda_cidade_atual;
        }
        somatorio_de_vendas += renda_cidade_atual;

        // verificando possivel maior e menor rendoa
        if (renda_cidade_atual >= maior_renda){
            maior_renda = renda_cidade_atual;
        }

        if (renda_cidade_atual <= menor_renda){
            menor_renda = renda_cidade_atual;
        }

        contador++;
    }

    // renda final média
    renda_media = somatorio_de_vendas / quantidade_cidades;

    // saída
    var resultado_final = `
    -------- resultados -------
    > renda média: R$ ${renda_media.toFixed(2)}
    > menor renda: R$ ${menor_renda}
    > maior renda: R$ ${maior_renda}\n`

    console.log(resultado_final)

}

main();

File: API_R_01Agosto2024/José Macêdo Dos Santos Junior - ADS/q3_cidades.py
# Feito por José Macêdo Dos Santos Junior
# Matrícula: 2024111TADS0038

#Função principal.
def main():
    # Variáveis
    renda_geral = 0
    menor_renda_cidade = 'não_definido'
    menor_renda = 'não_definido'
    maior_renda_cidade = 'não_definido'
    maior_renda = 'não_definido'

    # Entrada
    numero_cidades = int(input('> Informe o número de cidades que você irá informar a renda:\n> '))

    # Processamento
    contador = 1
    while contador < numero_cidades + 1:
        print(f'\n> --- Cidade [{contador}] de [{numero_cidades}]')
        cidade = str(input('> Informe a cidade:\n> ')).strip()
        renda = float(input(f'> Digite a renda de {cidade}:\n> '))
        if renda != 0.00:
            # Para a menor renda.
            if menor_renda == 'não_definido':
                menor_renda_cidade = cidade
                menor_renda = renda
            else:
                if menor_renda > renda:
                    menor_renda_cidade = cidade
                    menor_renda = renda
        
            # Para a maior renda.]
            if maior_renda == 'não_definido':
                maior_renda_cidade = cidade
                maior_renda = renda
            else:
                if maior_renda < renda:
                    maior_renda_cidade = cidade
                    maior_renda = renda
        else:
            # contador = float('+inf') Infinito positivo em Python.
            break
        renda_geral += renda
        contador += 1
    
    # Calculando a renda média.
    renda_media = renda_geral / numero_cidades

    # Mostrando na tela.
    print(f'''\n> --- RESULTADOS
> {'-' * 50}
> Cidade com maior renda ----- [{maior_renda_cidade}]
> renda da cidade ------------ [{maior_renda}]
> {'-' * 50}
> Cidade com menor renda ----- [{menor_renda_cidade}]
> renda da cidade------------- [{menor_renda}]
> {'-' * 50}
> Renda média ---------------- [{renda_media:.2f}]
> {'-' * 50}''')


if __name__ == '__main__':
    main()
